quic/s2n-quic-core/src/recovery/hybrid_slow_start.rs

```diff
@@ -64,6 +76,13 @@ impl HybridSlowStart {
         if congestion_window >= self.threshold {
             // We are already out of slow start so nothing to do
             return;
+        } else if self.use_hystart && self.threshold < f32::MAX {
+            // self.threshold has some values, which means we already performed hystart++ once.
+            // Fallback to nomal slow start as suggested in the hystart draft
+            //   "An implementation SHOULD use HyStart++ only for the initial slow start
+            //    and fall back to using traditional slow start for the remainder of the
+            //    connection lifetime." (Section 4.3 draft-ietf-tcpm-hystartplusplus-04)
```

Comment:
though I think the idea here is that once we have determined the slow start threshold (`self.threshold`), we don't need to run the slow start algorithm anymore. So maybe both these if checks could be simplified as just:
```rust
if self.threshold < f32:MAX {
    // We have already determined the slow start threshold so nothing to do
    return;
}
```

Reply:
yeah, you are right. The hybrid slow start paper seems to indicate we should re-attempt to find the slow start threshold after a timeout, I may have taken that to mean after persistent congestion as well, as that is the only case currently where we re-enter Slow Start after exiting it. 

Do you think there is harm is just running the hystart++ algorithm again when we've re-entered Slow Start after persistent congestion?

Reply:
I guess there will be,  although it's not significant. if we do hystart++ again after persistent congestion, its cwnd growth might become a bit slower when there're some jitters in RTTs.

Reply:
yeah, I suppose for hystart++, the whole idea is that you need the css phase (which grows cwnd slower) to determine the actual slow start threshold, but once you've already found that threshold there is no need to slow down the cwnd growth again the next time slow start begins. On the other hand though, I'm not sure how valid the slow start threshold found during the first run of hystart++ is after persistent congestion has been experienced. If there is persistent congestion, that seems to indicate something in the network path may have changed, so it might be worth running hystart++ again.

Reply:
yes, that's true. I believe hystart++ presumes that the possibilities for path changes is low. If we think this is not a good assumption, we can change this part.

Reply:
coming from reading the BBRv2 spec, which is constantly slowing down and speeding up to measure any changes in the path, I would say the assumption is probably not accurate. Either way though the negative consequences of being wrong are pretty low.